test/GPv2Settlement.test.ts

```diff
@@ -5,14 +5,20 @@ import { ethers, waffle } from "hardhat";
 import { domain } from "../src/ts";
 
 describe("GPv2Settlement", () => {
+  const [owner, solver] = waffle.provider.getWallets();
   let settlement: Contract;
+  let controller: Contract;
 
   beforeEach(async () => {
+    const GPv2AccessControl = await ethers.getContractFactory(
+      "GPv2AccessControl",
+    );
+    controller = await GPv2AccessControl.connect(owner).deploy();
+
     const GPv2Settlement = await ethers.getContractFactory(
       "GPv2SettlementTestInterface",
     );
-
-    settlement = await GPv2Settlement.deploy();
+    settlement = await GPv2Settlement.connect(owner).deploy(controller.address);
```

Comment:
I don't think `connect(owner)` is needed here - it suggests that the settlement contract needs to be deployed as a specific owner but I don't think that is true (and I don't think it should be true).

Reply:
Its not needed, but its is more precise. Otherwise the reader would have to know that the first account returned by `getWallets` is the default transaction sender.

Reply:
Right, sorry, I added this in the wrong place.